microcosm_fastapi/errors.py

```diff
@@ -15,4 +15,89 @@ class ErrorSchema(BaseSchema):
     message: str = "Unknown Error"
     code: int = 500
     retryable: bool
-    context: Optional[ErrorContextSchema]
\ No newline at end of file
+    context: Optional[ErrorContextSchema]
+
+
+class ParsedException:
+    def __init__(self):
+        self._error = None
+
+        self.context = None
+        self.retryable = False
+        self.error_message = None
+        self.status_code = 500
+        self.include_stack_trace = True
+
+    @property
+    def error(self):
+        return self._error
+
+    @error.setter
+    def error(self, value):
```

Comment:
NIT: It's unexpected to me that all attributes are set by a setter for one of them. I'd change it, so the `value` will be passed to `__init__`. This way we could rely on `ParsedException` instances that they always store actually parsed exception, not only if a setter was called.

There's just a question of how to change it's usage in `RequestInfo`, but maybe we could replace it with `None`, since there are no parsed exception when initialising request info?

Reply:
Yep nice improvement. Yeah I initialising to `None` would work much better and then setting the values on initialising of the ParsedException.